src/sinks/nats/config.rs

```diff
@@ -0,0 +1,130 @@
+use codecs::JsonSerializerConfig;
+use futures_util::TryFutureExt;
+use snafu::ResultExt;
+use vector_core::tls::TlsEnableableConfig;
+
+use crate::{
+    nats::{from_tls_auth_config, NatsAuthConfig, NatsConfigError},
+    sinks::prelude::*,
+};
+
+use super::{sink::NatsSink, ConfigSnafu, ConnectSnafu, NatsError};
+
+/// Configuration for the `nats` sink.
+#[configurable_component(sink(
+    "nats",
+    "Publish observability data to subjects on the NATS messaging system."
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct NatsSinkConfig {
+    #[configurable(derived)]
+    pub(super) encoding: EncodingConfig,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    pub acknowledgements: AcknowledgementsConfig,
+
+    /// A NATS [name][nats_connection_name] assigned to the NATS connection.
+    ///
+    /// [nats_connection_name]: https://docs.nats.io/using-nats/developer/connecting/name
+    #[serde(default = "default_name", alias = "name")]
+    #[configurable(metadata(docs::examples = "foo"))]
+    pub(super) connection_name: String,
+
+    /// The NATS [subject][nats_subject] to publish messages to.
+    ///
+    /// [nats_subject]: https://docs.nats.io/nats-concepts/subjects
+    #[configurable(metadata(docs::templateable))]
+    #[configurable(metadata(
+        docs::examples = "{{ host }}",
+        docs::examples = "foo",
+        docs::examples = "time.us.east",
+        docs::examples = "time.*.east",
+        docs::examples = "time.>",
+        docs::examples = ">"
+    ))]
+    pub(super) subject: String,
```

Comment:
Should we change this type to `Template`, rather than trying to convert it to a template when building the sink?

```suggestion
    pub(super) subject: Template,
```

Reply:
Good call.